state-and-event-handling/updating-state.md

```diff
@@ -0,0 +1,442 @@
+| `setPieceOfState(...)`                              | **Replace this** with the function that was created with `useState`. |
+| `'New value of pieceOfState.'`                      | **Replace this** with the new value for the piece of state.          |
+
+This code will update a piece of state named `pieceOfState`, and its new value will be `'New value of pieceOfState.'`.
+
+### Updating State Re-Renders the Component
+
+Recall that pieces of state are responsible for managing how one piece of the component looks. For example, they can control if an element shows or hides or the classes that an element has.
+
+**Every time a piece of state is updated, the component re-renders**.
+
+### !callout-warning
+
+## Working With State and Re-Rendering
+
+Re-rendering a component is hard work! Firstly, setting state is _asynchronous_. Secondly, to repeat, _every time that state updates, the component re-renders_. These two facts will aid us when debugging state.
+
+### !end-callout
+
+## Updating State As Part of Event Handling
+
+The best time to update state is in response to _handling user events_.
+
+We can use our update functions inside an event handler.
+
+### Example: Increasing Like Count
+
+Let's imagine a `Post` component. It's responsible for displaying the number of likes that a post has.
+
+Each post starts with zero likes. When we click the "like" button, the number of likes increases.
+
+Let's consider the code we need to write for this:
+
+1. The `Post` component should have a piece of state, `likesCount`.`likesCount` is ideally a piece of state, and not a `prop`, because state can be modified within a component.
+1. We should also get a function to update this piece of state, `setLikesCount`.
+1. The "like" button should listen for click events, using the attribute `onClick`.
+1. When the "like" button is clicked, we should call an event handler function. We can name this function `increaseLikes`.
+1. When our component increases likes in `increaseLikes`, we should update our state using `setLikesCount`.
+
+<!-- prettier-ignore-start -->
+```js
+import { useState } from "react";
+
+const Post = () => {
+    const [likesCount, setLikesCount] = useState(0);
+    const increaseLikes = () => {
+        console.log('We\'re inside increaseLikes!');
+        setLikesCount(likesCount + 1);
+    }
+
+    return (<section>
+        <p>The number of likes is {likesCount}.</p>
+        <button onClick={increaseLikes}>Like</button>
+    </section>)
+}
+
+export default Post;
+```
+<!-- prettier-ignore-end -->
+
+When we look at our web app, our count of likes starts at zero:
+
+![An app that reads "The number of likes is 0." and a "like" button. The browser Dev Tools are open.](../assets/state-and-event-handling_updating-state_zero-likes.png)  
+_Fig. An app that reads "The number of likes is 0."_
+
+When we click it once, we see that `increaseLikes` has been invoked... and the number of likes has increased! Our state has been updated!
```

Comment:
Since we haven't gotten to the image, it sounds better to me to have this be present.
explicit mention of state update leading to a re-render
point out that the initial value is still being passed on subsequent renders, but that it is ignored
```suggestion
When we click the "like" button once, we see that `increaseLikes` gets invoked... and the number of likes increases! Our state has been updated, and our component has been re-rendered.

Even though our `useState` is still passing `0` for the initial value, the value that `likesCount` gets on each following render is the updated value!
```